### PR TITLE
fix docker_build_opts bound variable

### LINKED ISSUE #82962: docker_build_opts bound variable
**What happened**:
when I run 
`KUBE_GIT_VERSION=v1.17.0-alpha.0 KUBE_FASTBUILD=true  KUBE_BUILD_PULL_LATEST_IMAGES=n make release-images`, found bound variable err.

```
+++ [0921 20:27:24] Verifying Prerequisites....
+++ [0921 20:27:24] Building Docker image kube-build:build-8eeb52dc0c-5-v1.12.9-1
+++ [0921 20:27:27] Creating data container kube-build-data-8eeb52dc0c-5-v1.12.9-1
   +++ [0921 20:27:36] Syncing sources to container
+++ [0921 20:27:46] Running build command...
   +++ [0921 20:27:55] Building go targets for linux/amd64:
    ./vendor/k8s.io/code-generator/cmd/deepcopy-gen
+++ [0921 20:28:05] Building go targets for linux/amd64:
    ./vendor/k8s.io/code-generator/cmd/defaulter-gen
+++ [0921 20:28:13] Building go targets for linux/amd64:
    ./vendor/k8s.io/code-generator/cmd/conversion-gen
+++ [0921 20:28:26] Building go targets for linux/amd64:
    ./vendor/k8s.io/kube-openapi/cmd/openapi-gen
+++ [0921 20:28:40] Building go targets for linux/amd64:
    ./vendor/github.com/go-bindata/go-bindata/go-bindata
+++ [0921 20:28:42] Building go targets for linux/amd64:
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kube-scheduler
    cmd/kube-proxy
    cmd/hyperkube
    vendor/github.com/onsi/ginkgo/ginkgo
    test/e2e/e2e.test
    cluster/images/conformance/go-runner
    cmd/kubectl




+++ [0921 20:32:17] Syncing out of container
+++ [0921 20:32:29] Building images: linux-amd64
+++ [0921 20:32:29] Starting docker build for image: kube-apiserver-amd64
+++ [0921 20:32:29] Starting docker build for image: kube-controller-manager-amd64
+++ [0921 20:32:29] Starting docker build for image: kube-scheduler-amd64
+++ [0921 20:32:29] Starting docker build for image: kube-proxy-amd64
/root/go/src/k8s.io/kubernetes/build/lib/release.sh:行386: docker_build_opts[@]: bound variable
+++ [0921 20:32:29] Building conformance image for arch: amd64
+++ [0921 20:32:29] Building hyperkube image for arch: amd64
/root/go/src/k8s.io/kubernetes/build/lib/release.sh:行386: docker_build_opts[@]: bound variable
/root/go/src/k8s.io/kubernetes/build/lib/release.sh:行386: docker_build_opts[@]: bound variable
/root/go/src/k8s.io/kubernetes/build/lib/release.sh:行386: docker_build_opts[@]: bound variable



+++ [0921 20:32:52] Deleting hyperkube image k8s.gcr.io/hyperkube-amd64:v1.17.0-alpha.0-dirty
+++ [0921 20:33:03] Deleting conformance image k8s.gcr.io/conformance-amd64:v1.17.0-alpha.0-dirty
!!! [0921 20:33:03] previous Docker build failed
!!! [0921 20:33:03] Call tree:
!!! [0921 20:33:03]  1: /root/go/src/k8s.io/kubernetes/build/lib/release.sh:231 kube::release::create_docker_images_for_server(...)
!!! [0921 20:33:03]  2: build/release-images.sh:42 kube::release::build_server_images(...)
make: *** [release-images] ERR 1
```
**What you expected to happen**:

Build success

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**: 
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):Linux paasn1 4.14.78-300.el7.Centos.x86_64 #1 SMP Sat Jan 5 15:49:43 CST 2019 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:
GNU bash， version 4.2.46(2)-release (x86_64-redhat-linux-gnu)
Copyright (C) 2011 Free Software Foundation, Inc.

### PR DESCRIPTION
**Pull request by zouyee**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
fix docker_build_opts bound variable

**Which issue(s) this PR fixes**:

Fixes #82962

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by neolit123** [comment]
@zouyee 
please make sure the CI tests pass.

/uncc

**Comment by zouyee** [comment]
@cblecker  PTAL

File: build/lib/release.sh [hunk 1]
```diff
@@ -345,6 +345,15 @@ function kube::release::create_docker_images_for_server() {
       return 1
     fi
 
+    # provide `--pull` argument to `docker build` if `KUBE_BUILD_PULL_LATEST_IMAGES`
+    # is set to y or Y; otherwise try to build the image without forcefully
+    # pulling the latest base image.
+    local DOCKER_BUILD_OPTS=()
+    if [[ "${KUBE_BUILD_PULL_LATEST_IMAGES}" =~ [yY] ]]; then
```
Comment (fejta) [inline build/lib/release.sh]: Does something like this work?
```bash
if [[ pull == ~y ]]; then
  local -r docker_build_opts=(--pull)
else
  local -r docker_build_opts=()
fi
```

**Review by fejta — COMMENTED** [review]
/lgtm
/approve
/hold

File: build/lib/release.sh (hunk 1 above)
Comment (zouyee) [reply build/lib/release.sh]: After testing, it does not work.

**Pull request merged**